File: sol/solution/8 Kyu/56efc695740d30f963000557/Solution.dart
/*
* Solution
*/
String toAlternatingCase(final String str) {
  final buffer = StringBuffer();
  for (var codeUnit in str.codeUnits) {
    var c = String.fromCharCode(codeUnit);
    if (c.toUpperCase() != c && c.toLowerCase() == c) {
      buffer.write(c.toUpperCase());
    } else if (c.toLowerCase() != c && c.toUpperCase() == c) {
      buffer.write(c.toLowerCase());
    } else {
      buffer.write(c);
    }
  }
  return buffer.toString();
}





/*
* Sample Tests
*/
// See https://pub.dartlang.org/packages/test
import "package:test/test.dart";
import "package:solution/solution.dart";

void main() {
  group("Sample Test Cases", () {
    test("values: 'hello world'; expected result: 'HELLO WORLD'", () => expect(toAlternatingCase("hello world"), equals("HELLO WORLD")));
    test("values: 'HeLLo WoRLD'; expected result: 'hEllO wOrld'", () => expect(toAlternatingCase("HeLLo WoRLD"), equals("hEllO wOrld")));
    test("values: '1a2b3c4d5e'; expected result: '1A2B3C4D5E'", () => expect(toAlternatingCase("1a2b3c4d5e"), equals("1A2B3C4D5E")));
    test("values: '12345'; expected result: '12345'", () => expect(toAlternatingCase("12345"), equals("12345")));
    test("values: 'String.prototype.toAlternatingCase'; expected result: 'sTRING.PROTOTYPE.TOaLTERNATINGcASE'", () => expect(toAlternatingCase("String.prototype.toAlternatingCase"), equals("sTRING.PROTOTYPE.TOaLTERNATINGcASE")));
  });
}

File: sol/solution/8 Kyu/56efc695740d30f963000557/Solution.java
/**
 * Solution
**/
public class StringUtils {
  public static String toAlternativeString(String str) {
    StringBuilder result = new StringBuilder();
    for (char c : str.toCharArray()) {
      if (Character.isLowerCase(c)) {
        result.append(Character.toUpperCase(c));
      } else if (Character.isUpperCase(c)) {
        result.append(Character.toLowerCase(c));
      } else {
        result.append(c);
      }
    }
    return result.toString();
  }
}





/**
* Sample Tests
**/
import org.junit.Test;

import static org.junit.Assert.*;

public class StringUtilsTest {

  @Test
  public void fixedTests() {
    assertEquals("HELLO WORLD", StringUtils.toAlternativeString("hello world"));
    assertEquals("hello world", StringUtils.toAlternativeString("HELLO WORLD"));
    assertEquals("HELLO world", StringUtils.toAlternativeString("hello WORLD"));
    assertEquals("hEllO wOrld", StringUtils.toAlternativeString("HeLLo WoRLD"));
    assertEquals("Hello World", StringUtils.toAlternativeString(StringUtils.toAlternativeString("Hello World")));
    assertEquals("12345", StringUtils.toAlternativeString("12345"));
    assertEquals("1A2B3C4D5E", StringUtils.toAlternativeString("1a2b3c4d5e"));
    assertEquals("sTRINGuTILS.TOaLTERNATINGcASE", StringUtils.toAlternativeString("StringUtils.toAlternatingCase"));
  }
  
  @Test
  public void kataTitleTests() {
    assertEquals("ALTerNAtiNG CaSe", StringUtils.toAlternativeString("altERnaTIng cAsE"));
    assertEquals("altERnaTIng cAsE", StringUtils.toAlternativeString("ALTerNAtiNG CaSe"));
    assertEquals("ALTerNAtiNG CaSe <=> altERnaTIng cAsE", StringUtils.toAlternativeString("altERnaTIng cAsE <=> ALTerNAtiNG CaSe"));
    assertEquals("altERnaTIng cAsE <=> ALTerNAtiNG CaSe", StringUtils.toAlternativeString("ALTerNAtiNG CaSe <=> altERnaTIng cAsE"));
  }
}